support-lib/jni/djinni_support.cpp

```diff
@@ -239,8 +239,13 @@ jint JniEnum::ordinal(JNIEnv * env, jobject obj) const {
 
 LocalRef<jobject> JniEnum::create(JNIEnv * env, jint value) const {
     LocalRef<jobject> values(env, env->CallStaticObjectMethod(m_clazz.get(), m_staticmethValues));
+    jniExceptionCheck(env);
     DJINNI_ASSERT(values, env);
-    return LocalRef<jobject>(env, env->GetObjectArrayElement(static_cast<jobjectArray>(values.get()), value));
+    LocalRef<jobject> result(env,
+                             env->GetObjectArrayElement(static_cast<jobjectArray>(values.get()),
+                                                        value));
+    jniExceptionCheck(env);
+    return std::move(result);
```

Comment:
```cpp
X foo() { const X x; return x; }
int main() {
  X x = foo();
  x.mutate();
}
```
In this case the variable `x` is declared as `const` in the callee. If it used RVO then accessing that variable as non-`const` after the call were UB since you are removing the `const`-ness.

```cpp
X foo() { X x; return x; }
int main() {
  const X x = foo();
  x.access();
}
```
Here `x` is created as non-`const` via RVO and then merely used as `const` in the caller. This is like passing a variable as `const&` argument, but applied locally in the caller.